sklearn/preprocessing/imputation.py

```diff
@@ -357,6 +379,18 @@ def transform(self, X):
 
             X.data[mask] = astype(valid_statistics[indexes], X.dtype,
                                   copy=False)
+
+            if self.add_missing_indicator:
+                mask_matrix = X.__class__((mask, X.indices.copy(),
+                                           X.indptr.copy()), shape=X.shape, dtype=X.dtype)
+                mask_matrix.eliminate_zeros()  # removes explicit False entries to make the matrix sparser
+                features_with_missing_values = np.where(sparse_min_max(mask_matrix, 0)[1])[0]
```

Comment:
``sparse_min_or_max`` is returning a tuple of ``(min,max)`` where each of them is a 1D row vector.